arex-instrumentation/redis/arex-spring-data-redis/src/main/java/io/arex/inst/spring/data/redis/SpringDataRedisInstrumentation.java

```diff
@@ -0,0 +1,230 @@
+package io.arex.inst.spring.data.redis;
+
+import static java.util.Arrays.asList;
+import static net.bytebuddy.matcher.ElementMatchers.namedOneOf;
+import io.arex.agent.bootstrap.model.MockResult;
+import io.arex.inst.extension.MethodInstrumentation;
+import io.arex.inst.extension.TypeInstrumentation;
+import io.arex.inst.redis.common.RedisKeyUtil;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.data.redis.core.RedisTemplate;
+
+/**
+ * SpringDataRedisInstrumentation
+ */
+public class SpringDataRedisInstrumentation extends TypeInstrumentation {
+
+    @Override
+    protected ElementMatcher<TypeDescription> typeMatcher() {
+        return namedOneOf("org.springframework.data.redis.core.DefaultValueOperations",
+            "org.springframework.data.redis.core.DefaultSetOperations",
+            "org.springframework.data.redis.core.DefaultZSetOperations",
+            "org.springframework.data.redis.core.DefaultListOperations",
+            "org.springframework.data.redis.core.DefaultHashOperations");
+    }
+
+    @Override
+    public List<MethodInstrumentation> methodAdvices() {
+        return asList(MethodCollector.arg1IsObjectKey(Arg1IsObjectKeyAdvice.class.getName()),
+            MethodCollector.arg1IsCollectionKey(Arg1IsCollectionKeyAdvice.class.getName()),
+            MethodCollector.arg1IsMapKey(Arg1IsMapKeyAdvice.class.getName()),
+            MethodCollector.arg1IsObjectKeyArg2IsCollectionKey(
+                Arg1IsObjectKeyArg2IsCollectionKeyAdvice.class.getName()),
+            MethodCollector.arg1IsObjectKeyArg2IsMapKey(Arg1IsObjectKeyArg2IsMapKeyAdvice.class.getName()),
+            MethodCollector.arg1AndArg2AreObjectKey(Arg1AndArg2AreObjectKeyAdvice.class.getName()));
+    }
+
+    public static class Arg1IsObjectKeyAdvice {
```

Comment:
If it is just because the key types are inconsistent, there is no need to distinguish, just make the determination in RedisKeyUtil.generate(xx).

Reply:
fixed